### PR TITLE
Add a Honeycomb exporter

### PR DESCRIPTION
**Pull request by paulosman**
Add a honeycomb exporter so users of the collector can send traces to their Honeycomb account. This uses the opentelemetry Go SDK [Honeycomb exporter package](https://github.com/honeycombio/opentelemetry-exporter-go), which uses [libhoney-go](https://github.com/honeycombio/libhoney-go) under the hood to send trace data as batched events.

### CODE REVIEW

**Comment by paulosman** [comment]
Tests were passing locally but failing CI. Investigating..

File: exporter/honeycombexporter/honeycomb.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	hce := HoneycombExporter{
+		exporter: exporter,
+	}
+	return exporterhelper.NewTraceExporter(
+		cfg,
+		hce.pushTraceData,
+		exporterhelper.WithTracing(true),
+		exporterhelper.WithMetrics(false),
```
Comment (pjanotti) [inline exporter/honeycombexporter/honeycomb.go]: this is an option to generate metrics for the exporter, it should be set to true. BTW, we are removing these options and settings the metrics to be "on" by default, but, no worries I will take care of that when it reaches the contrib repo

File: exporter/honeycombexporter/honeycomb.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/honeycombio/opentelemetry-exporter-go/honeycomb"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	"go.opentelemetry.io/otel/api/core"
+)
+
+const oTelCollectorUserAgentStr = "Honeycomb-OpenTelemetry-collector"
```
Comment (pjanotti) [inline exporter/honeycombexporter/honeycomb.go]:
```suggestion
const oTelCollectorUserAgentStr = "Honeycomb-OpenTelemetry-Collector"
```
?

File: go.mod [hunk 3]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter v0.0.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter v0.0.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/carbonexporter v0.0.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/honeycombexporter v0.0.0-00010101000000-000000000000
```
Comment (pjanotti) [inline go.mod]: nit: no practical difference, just to match the others
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/honeycombexporter v0.0.0
```

**Review by pjanotti — APPROVED** [review]
LGTM @paulosman, just a few nits.

> which uses libhoney-go under the hood to send trace data as batched events.

Notice that typically we recommend sending the batch as it is without queueing or other steps on the exporter for consistency in composability and observability but it makes sense to be as it is to allow code re-use.

File: exporter/honeycombexporter/honeycomb.go (hunk 1 above)
Comment (paulosman) [reply exporter/honeycombexporter/honeycomb.go]: > this is an option to generate metrics for the exporter, it should be set to true. BTW, we are removing these options and settings the metrics to be "on" by default, but, no worries I will take care of that when it reaches the contrib repo

I'll push a commit in a moment with the suggestions below, so might as well get this one too. Thanks!

File: exporter/honeycombexporter/honeycomb.go (hunk 2 above)
Comment (paulosman) [reply exporter/honeycombexporter/honeycomb.go]: Good catch, changed in fa4961a732475b95b6a11705319077519d45dec8

File: go.mod (hunk 3 above)
Comment (paulosman) [reply go.mod]: Perfect - fixed in fa4961a732475b95b6a11705319077519d45dec8. Thanks!

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by lizthegrey — APPROVED** [review]

**Pull request merged**